src/rgw/rgw_rest_swift.cc

```diff
@@ -282,15 +282,17 @@ void RGWListBucket_ObjStore_SWIFT::send_response()
   while (iter != objs.end() || pref_iter != common_prefixes.end()) {
     bool do_pref = false;
     bool do_objs = false;
-    rgw_obj_key& key = iter->key;
+    if (iter != objs.end()) {
+      rgw_obj_key& key = iter->key;
+    }
```

Comment:
@yehudasa See question I asked you at https://github.com/ceph/ceph/commit/5cf5ab470e6e9119ae4a6a6111974cb03610ba3e